### DISCUSSION TITLE
trees near ML topology

### DISCUSSION THREAD

**Discussion by diploid2n** [iqtree/iqtree2#268] [Q&A]
Hi,

I am wondering if it is possible to collect tree topologies which are not final ml tree but intermediate ml tree during each round of topology search.   

Thanks in advance.

Tae-Kun Seo

**Comment by roblanf** [comment]
I think the option you're looking for is:

`-wt`: Turn on writing all locally optimal trees into .treels file.

Rob

**Reply by diploid2n** [reply]
Thanks!  
Are those trees listed in order of log-likelihood score?  For example, if I need best 10 alternative tree topologies, then can I collect last 10 topologies?

**Reply by roblanf** [reply]
Sorry, no idea (@bqminh is on leave right now, but he'll know when he's back). One option would be to calculate the likelihood of all of them by fixing the topology and branch lengths, and your model (but not the model parameters). This is imperfect, because you can't guarantee you'll get exactly the same likelihood if the optimiser finds slightly different optimal model parameters, but it should be pretty reliable.

**Reply by jaredshemon** [reply]
Did you ever figure this out? I'm looking for the top log likelihood scores as well.

**Comment by bqminh** [comment]
Sorry I didn't notice this question is for me.

`-wt` print trees occurred during tree search, not sorted by log-likelihood.

If you really want to see the top k trees by likelihood: They are stored in the checkpoint file .ckp.gz, in the section "CandidateSet", for example:

```
CandidateSet:
 00: -23646.0180312 (0:0.1169020346,(1:0.1176852561,2:0.1239009854):0.07155698657,(3:0.1543000497,((4:0.1359919501,((5:0.1790154682,6:0.2000157859):0.02763066206,(7:0.1672651305,8:0.1467018792):0.06282203328):0.03541521671):0.04318519401,(((9:0.1362320039,(10:0.0755849736,(11:0.0683464242,12:0.0813507866):0.02843657614):0.02261438764):0.02815089397,(13:0.0545375188,14:0.0701529349):0.07373297132):0.0386179566,(15:0.1311729590,16:0.1100511848):0.03811696936):0.07882431445):0.07990050469):0.05524666025);
 01: -23650.1276455 (0:0.1172247159,(1:0.1172523864,2:0.1240084208):0.07153680156,(3:0.1555534264,((4:0.1355701935,((5:0.1791894152,6:0.1999345288):0.02712004633,(7:0.1672883666,8:0.1466539611):0.06335581075):0.03609976388):0.04423487761,((((9:0.1402574031,(11:0.0681616954,12:0.0807670791):0.0246100167):0.01986292641,10:0.0734764911):0.03615647584,(13:0.0518957662,14:0.0727795604):0.07248176629):0.03795587573,(15:0.1320279167,16:0.1087036346):0.0374839695):0.07906545914):0.07847360379):0.05461054602);
 02: -23652.1889613 (0:0.1165146228,(1:0.1171770373,2:0.1241530041):0.07182298964,(3:0.1557754313,((4:0.1370857901,((5:0.1875360531,(7:0.1665774936,8:0.1466962265):0.06072968694):0.01805458417,6:0.2113734219):0.03533002468):0.04261557786,(((9:0.1363003950,(10:0.0751942604,(11:0.0681574322,12:0.0815209448):0.02859101755):0.02248251308):0.02923442017,(13:0.0538556254,14:0.0705252110):0.0738703991):0.0376589181,(15:0.1313981539,16:0.1101250457):0.03868605324):0.0760996041):0.07990389691):0.05497404474);
```

Each line within this section shows the ID, log-likelihood and the tree string, with taxon ID instead of taxon names. The trees are sorted, with tree 00 having the highest log-likelihood.

Does that answer your question?

**Reply by jaredshemon** [reply]
Yes that answered my question and helped my analysis, thank you!
